native/src/main/java/io/ballerina/stdlib/ftp/util/FtpContentConverter.java

```diff
@@ -150,9 +148,19 @@ public static Object convertBytesToCsv(byte[] content, Type targetType) {
      * @return BMap containing parse options
      */
     private static BMap<BString, Object> createJsonParseOptions() {
-        BMap<BString, Object> options = ValueCreator.createMapValue();
-        // Enable flexible data projection
-        options.put(ALLOW_DATA_PROJECTION, true);
+        BMap<BString, Object> options = ValueCreator.createRecordValue(
```

Comment:
Could you add a bit of detail on the logic here? I'm reading it as jsondata:parseOption being created, and then its default value is immediately checked against BMap.

Should this be a configurable value? If so, would it be better to expose it in the ListenerConfiguration to be set during config processing?